sycl/test/basic_tests/free_function_queries/free_function_queries_interface.cpp

```diff
@@ -53,7 +53,7 @@ template <template <int, bool = true> class Item, int Dims> void test() {
       "Wrong return type of free function query for Item");
 }
 
-int main() {
+void test_all() {
```

Comment:
```suggestion
SYCL_EXTERNAL void test_all() {
```
I think adding `SYCL_EXTERNAL` to `test_all` declaration is enough to remove `main` function.

Reply:
I am not sure I follow. How so?

Reply:
According to my understanding, the goal of the test changes is to make sure that `test_all` function is compiled by both host and device compilers. Right?
The same can be achieved with less code - `SYCL_EXTERNAL` functions are compiled by both compilers.

The code of this test is not linked, so `main` function is not necessary.

Reply:
I see what you mean. Yeah, that absolutely works. I have made the corresponding changes.

Reply:
Thanks!